### PR TITLE
feat: Updating filters triggers animation

### PR DESCRIPTION
**Pull request by arnest00**
### Description
- TradingCardGrid fades in on inital load
- TradingCardGrid fades out and then in with each filter selection

### Spec
Designs: [Designs](https://www.figma.com/file/kswpPSHH2h5njVyfGDbIHk/FSA-Capstone-Project-2022)
References: [247](https://sparkbox.atlassian.net/browse/FSA22V1-247)

### Validation
* [ ] This PR has visual elements, so it was reviewed by a designer.
* [ ] This PR has code changes, and our linters still pass.
* [ ] This PR affects production code, so it was browser tested.
* [ ] Changes are browser tested. This includes functionality, accessibility, responsiveness, and design.

#### To Validate
1. Make sure all PR Checks have passed.
2. Pull down `feat--update-animation` branch.
3. Run `npm i`.
4. Confirm all tests pass.
5. Confirm all linters pass.
6. Open deploy preview and verify expected behavior on Overview page.

### CODE REVIEW

**Review by mdt2 — DISMISSED** [review]
qYGvebgOKGygdQgflY

**Review by theogainey — CHANGES_REQUESTED** [review]
Overall good work. Animations are fun!  But two little things, 1) can we a `prefers-reduced-motion` query so that the animations are off for those who need or want reduced motion. 2) have you thought how the animation will look when the cards are in list view? If it will help I can get at least a draft PR up soon.

**Comment by arnest00** [comment]
Good looking out, @theogainey. The default animation is now the version with less motion, though if a user has no preference, then the version with more motion overrides the default.

And yes, I think this animation would work well with the list view, but once you get that draft PR up, I can verify with sight over imagination.

File: components/TradingCardGrid.jsx [hunk 1]
```diff
@@ -45,9 +93,15 @@ export default function TradingCardGrid({ teamMembers }) {
...
-        <ViewToggle />
+        <ViewToggle
+          currentView={view}
+          onChangeView={handleChangeView}
+        />
       </div>
-      <div className="cmp-trading-card-grid__grid">
+      <div className={`cmp-trading-card-grid__grid ${gridClassModifier}`}>
+        <Show when={!filteredTeamMembers.length}>
+          <p className="cmp-trading-card-grid__status">No team members fit your criteria.</p>
```
Comment (theogainey) [inline components/TradingCardGrid.jsx:103]: This line is redundant now that we have the `TradingCardNumResults` component. "0 your X meet your criteria" is plenty to let people know they no one matches

**Review by theogainey — CHANGES_REQUESTED** [review]
Big fan of the new fade in/out animation. Initially seeing  3 new `useEffects`, including one with a linting rule disabled, was a bit worrisome but after looking at this code for a while I think this code is good to go. This animation clearly wasn't the easiest thing to implement so nicely done. Just take a look at my comment about the redundant line and I'll happily call this a job well done.

**Review by theogainey — APPROVED** [review]

**Pull request merged**